### PR TITLE
Required Lifecycle Probes Policy

### PR DESCRIPTION
**Pull request by EmandM**

### CODE REVIEW

File: Policies/Kubernetes/container-required-probes/template.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sazurecontainerprobesrequired
+
+        violation[{"msg": msg}] {
+          container := input_containers[_]
+          probe := input.parameters.requiredProbes[_]
+          probe_is_missing(container, probe)
+          msg := sprintf("Container %v has is missing required probe %v. Required probes: %v", [container.name, probe, input.parameters.requiredProbes])
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/template.yaml:26]: >has [](start = 39, length = 3)

nit: remove 'has'

File: Policies/Kubernetes/container-required-probes/azurepolicy.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
+  "properties": {
+    "displayName": "Ensure all pods have lifecycle probes configured",
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/azurepolicy.json:3]: >Ensure all pods have lifecycle probes configured [](start = 20, length = 48)

Change to "Kubernetes cluster Pods should configure lifecycle probes"

File: Policies/Kubernetes/container-required-probes/azurepolicy.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "policyType": "BuiltIn",
+    "mode": "Microsoft.Kubernetes.Data",
+    "description": "This policy enforces that all pods have the expected lifecycle probes configured. This policy is generally available for Kubernetes Service (AKS), and preview for AKS Engine and Azure Arc enabled Kubernetes. For instructions on using this policy, visit https://aka.ms/kubepolicydoc.",
+    "metadata": {
+      "version": "1.0.1",
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/azurepolicy.json:8]: >version [](start = 7, length = 7)

Update version and add policy as preview

File: Policies/Kubernetes/container-required-probes/azurepolicy.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{
+  "properties": {
+    "displayName": "Ensure all pods have lifecycle probes configured",
+    "policyType": "BuiltIn",
+    "mode": "Microsoft.Kubernetes.Data",
+    "description": "This policy enforces that all pods have the expected lifecycle probes configured. This policy is generally available for Kubernetes Service (AKS), and preview for AKS Engine and Azure Arc enabled Kubernetes. For instructions on using this policy, visit https://aka.ms/kubepolicydoc.",
+    "metadata": {
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/azurepolicy.json:7]: Remove text "This policy is generally available for Kubernetes Service (AKS), and preview for AKS Engine and Azure Arc enabled Kubernetes" as we release this policy initially in preview mode.

**Comment by RamyasreeChakka** [comment]
@DCtheGeek We are planning to make this policy as built-in. Can you please review policy text?

File: Policies/Kubernetes/container-required-probes/azurepolicy.json [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    },
+    "parameters": {
+      "requiredProbesList": {
+        "type": "Array",
+        "metadata": {
+          "displayName": "Required probes list",
+          "description": "The list of probes that are required to be defined on a container. Kubernetes currently supports 'livenessProbe', 'readinessProbe', and 'startupProbe'."
+        },
+        "defaultValue": ["readinessProbe", "livenessProbe"]
+      },
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/azurepolicy.json:19]: Can you add allowedValues  -> "readinessProbe", "livenessProbe", "startupProbe"

File: Policies/Kubernetes/container-required-probes/rego/src.rego [hunk 6]
```diff
@@ -0,0 +1,22 @@
+package k8sazurecontainerprobesrequired
+
+violation[{"msg": msg}] {
+  container := input_containers[_]
+  probe := input.parameters.requiredProbes[_]
+  probe_is_missing(container, probe)
+  msg := sprintf("Container %v has is missing required probe %v. Required probes: %v", [container.name, probe, input.parameters.requiredProbes])
```
Comment (RamyasreeChakka) [inline Policies/Kubernetes/container-required-probes/rego/src.rego:7]: nit: remove has

**Review by RamyasreeChakka — CHANGES_REQUESTED** [review]
Minor comments

**Review by denniszielke — APPROVED** [review]
LGTM

**Comment by mrajess** [comment]
Should this be closed out?

**Pull request closed without merging**